app/controllers/assessment360_controller.rb

```diff
@@ -59,21 +57,27 @@ def all_students_all_reviews
                                                    @meta_review_info_per_stu)
       end
       # calculate average grade for each student on all assignments in this course
-      if @teammate_review_info_per_stu[1] > 0
-        temp_avg_grade = @teammate_review_info_per_stu[0] * 1.0 / @teammate_review_info_per_stu[1]
-        @teammate_review[cp.id][:avg_grade_for_assgt] = temp_avg_grade.round.to_s + '%'
-      end
-      if @meta_review_info_per_stu[1] > 0
-        temp_avg_grade = @meta_review_info_per_stu[0] * 1.0 / @meta_review_info_per_stu[1]
-        @meta_review[cp.id][:avg_grade_for_assgt] = temp_avg_grade.round.to_s + '%'
-      end
+      review_info_per_student(cp,@teammate_review_info_per_stu, @teammate_review)
+      review_info_per_student(cp,@meta_review_info_per_stu, @meta_review)
     end
     # avoid divide by zero error
-    @assignments.each do |assignment|
-      temp_count = @overall_teammate_review_count[assignment.id]
-      @overall_teammate_review_count[assignment.id] = 1 if temp_count.nil? or temp_count.zero?
-      temp_count = @overall_meta_review_count[assignment.id]
-      @overall_meta_review_count[assignment.id] = 1 if temp_count.nil? or temp_count.zero?
+    avoid_divide_by_zero_error(@assignments,@overall_teammate_review_count,@overall_meta_review_count)
+  end
+
+  def avoid_divide_by_zero_error (assignments,overall_teammate_review_count,overall_meta_review_count)  
+    assignments.each do |assignment|
+      temp_count = overall_teammate_review_count[assignment.id]
+      overall_review_count_hash = 1 if temp_count.nil? or temp_count.zero?
+      temp_count = overall_meta_review_count[assignment.id]
+      overall_meta_review_count[assignment.id] = 1 if temp_count.nil? or temp_count.zero?
+    end
+  end
+
+
+  def review_info_per_student(cp,review_info_per_stu,review)
```

Comment:
Even if the data structure is turned into a hash, you would still need a comment describing what it is used for.  I'm sure it's taken you awhile to figure it out.  Don't make the next reader redo all that work.